src/test/java/org/wahlzeit_revisited/auth/AuthenticationFilterTest.java

```diff
@@ -0,0 +1,322 @@
+        Map<String, String> authHeader = Map.of(HttpHeaders.AUTHORIZATION, "Bearer " + encodedCredentials);
+        MockRequest mockRequest = new MockRequest(authHeader);
+
+        // execute
+        filter.filter(mockRequest);
+
+        // validate
+        Assert.assertNull(mockRequest.getAbortResponse());
+        Assert.assertNotNull(mockRequest.getSecurityContext());
+    }
+
+    @Test
+    @RolesAllowed(AccessRights.ADMINISTRATOR_ROLE)
+    public void test_InsufficientRights() throws NoSuchMethodException, SQLException {
+        // prepare
+        Method mockMethod = AuthenticationFilterTest.class.getMethod("test_InsufficientRights");
+        filter.resourceInfo = new MockResourceInfo(mockMethod);
+
+        // create new unique User
+        String email = buildUniqueEmail("insufficient");
+        String password = "SimplePasswort1234";
+        User authUser = userFactory.createUser("", email, password, AccessRights.USER);
+        userRepository.insert(authUser);
+
+        // add header according to the BASIC auth standard
+        String credentials = email + ":" + password;
+        String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+        Map<String, String> authHeader = Map.of(HttpHeaders.AUTHORIZATION, "Bearer " + encodedCredentials);
+        MockRequest mockRequest = new MockRequest(authHeader);
+
+        // execute
+        filter.filter(mockRequest);
+
+        // validate
+        Assert.assertNotNull(mockRequest.getAbortResponse());
+        Assert.assertNull(mockRequest.getSecurityContext());
+    }
+
+    @Test
+    @RolesAllowed(AccessRights.GUEST_ROLE)
+    public void test_SuperiorRightsFilter() throws NoSuchMethodException, SQLException {
+        // prepare
+        Method mockMethod = AuthenticationFilterTest.class.getMethod("test_SuperiorRightsFilter");
+        filter.resourceInfo = new MockResourceInfo(mockMethod);
+
+        // create new unique User
+        String email = buildUniqueEmail("superior");
+        String password = "SimplePasswort1234";
+        User authUser = userFactory.createUser("", email, password, AccessRights.USER);
+        userRepository.insert(authUser);
+
+        // add header according to the BASIC auth standard
+        String credentials = email + ":" + password;
+        String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+        Map<String, String> authHeader = Map.of(HttpHeaders.AUTHORIZATION, "Bearer " + encodedCredentials);
+        MockRequest mockRequest = new MockRequest(authHeader);
+
+        // execute
+        filter.filter(mockRequest);
+
+        // validate
+        Assert.assertNull(mockRequest.getAbortResponse());
+        Assert.assertNotNull(mockRequest.getSecurityContext());
+    }
+
+
+    /*
+     * Mock classes
+     */
+
+    private static class MockResourceInfo implements ResourceInfo {
```

Comment:
Does it make sense to use a library like `mockito` here instead?

Reply:
I totally agree, I just was uncertain about adding an other dependency